### DISCUSSION TITLE
Parsing genesis.json for gaia hub 4 seems oddly slow

### DISCUSSION THREAD

**Discussion by faddat** [tendermint/tendermint#6154] [General]
I did a bit of research and I am stumped.

128gb 12 core hetzner bare metal beast:

![image](https://user-images.githubusercontent.com/7142025/108590041-10414880-7394-11eb-9fb8-2db1ec6ef229.png)

....performance of ways to pull json into go apps, from here:

![image](https://user-images.githubusercontent.com/7142025/108522176-1df8be80-72ff-11eb-8026-0fddb0e3696f.png)

I found what I think is the function where genesis is loaded into the database:

**https://sourcegraph.com/github.com/tendermint/tendermint@b6be889b97598ed0bdd662e654705a411cab3940/-/blob/node/node.go#L1668**

I also found where we unmarshal json:

https://github.com/tendermint/tendermint/tree/master/libs/json

It doesn't seem exotic, so I am wondering where the bottleneck might be

**Comment by zmanian** [comment]
This is not a tendermint issue.

cosmos/cosmos-sdk#7766
